syne_tune/blackbox_repository/blackbox_surrogate.py

```diff
@@ -46,57 +53,155 @@ def __init__(
         X: pd.DataFrame,
         y: pd.DataFrame,
         configuration_space: dict,
+        objectives_names: List[str],
         fidelity_space: Optional[dict] = None,
         fidelity_values: Optional[np.array] = None,
         surrogate=None,
+        predict_curves: bool = False,
+        num_seeds: int = 1,
+        fit_differences: Optional[List[str]] = None,
         max_fit_samples: Optional[int] = None,
         name: Optional[str] = None,
     ):
         """
         Fits a blackbox surrogates that can be evaluated anywhere, which can be useful for supporting
         interpolation/extrapolation. To wrap an existing blackbox with a surrogate estimator, use `add_surrogate`
         which automatically extract X, y matrices from available blackbox evaluations.
-        :param X: dataframe containing hyperparameters values, columns should be the ones in configuration_space
-        and fidelity_space
-        :param y: dataframe containing objectives values
+
+        The surrogate regression model is provided by `surrogate`, it has to
+        conform to the scikit-learn fit-predict API. If `predict_curves` is True,
+        the model maps features of the configuration to the whole curve over
+        fidelities, separate for each metric and seed. This has several advantages.
+        First, predictions are consistent: if all curves in the data respect a certain
+        property which is retained under convex combinations, predictions have this
+        property as well (examples: positivity, monotonicity). This is important for
+        `elapsed_time` metrics. The regression models are also fairly compact, and
+        prediction is fast, `max_fit_samples` is normally not needed.
+
+        If `predict_curves` is False, the model maps features from configuration and
```

Comment:
This is the previous default. I switched predict_curves=True, because for k-nn, this runs much faster and provides the same results. For other surrogates, this is also much faster, but can of course also work worse.